### PR TITLE
azure_storage_container

### PR DESCRIPTION
**Pull request by enes-oezdemir**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Container level sas tokens on azure are generated without "?"
storage level sastokens however are generated with "?"

SAS Token without "?" doesn't work

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by YuvHayun — COMMENTED** [review]
So if I understand correctly this should fix the cases where an **account_sas_token** is given but doesn't contain '?' at the beginning?

**Comment by enes-oezdemir** [comment]
> So if I understand correctly this should fix the cases where an **account_sas_token** is given but doesn't contain '?' at the beginning?

yes

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/blob.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+xsoar
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/blob.txt:1]: what's this?

File: Packs/AzureStorageContainer/ReleaseNotes/1_0_20.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Storage Container
+
+- Fixed an issue where a sas token generated at container level didn't work properly.
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/ReleaseNotes/1_0_20.md:6]:
```suggestion
- Fixed an issue where a **sas token** generated at container level didn't work properly.
```

File: Packs/AzureStorageContainer/ReleaseNotes/1_0_20.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Storage Container
+
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/ReleaseNotes/1_0_20.md:5]: Note that you need to update about the docker update in the release notes as well.

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py [hunk 4]
```diff
@@ -976,6 +976,7 @@ def generate_sas_token_command(client: Client, args: dict) -> CommandResults:  #
         return result
     else:
         return_error("Permissions are invalid or in wrong order. Correct order for permissions are \'racwdl\'")
+        return None
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py:979]:
```suggestion
        raise DemistoException("Permissions are invalid or in wrong order. Correct order for permissions are \'racwdl\'")
```

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py [hunk 5]
```diff
@@ -999,6 +1000,7 @@ def test_module(client: Client) -> None:
         raise exception
 
     return_results('ok')
+    return None
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py:1003]:
```suggestion
```
why?

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
 import hashlib
 import hmac
 import shutil
-from typing import Callable
+from collections.abc import Callable
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py:6]: why changing?

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @enes-oezdemir thx for you contribution, nice catch!
Note that when updating an API module, you need to update all the integrations relying on this API module.
This can be done easily by running `demisto-sdk update-release-notes -g`.

However, if the fix is only relevant to this specific integration, then let's revert the changes to the API module and create a proper fix in the integration code.

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py (hunk 5 above)
Comment (enes-oezdemir) [reply Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py]: not me :) that's pre-commit

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py (hunk 6 above)
Comment (enes-oezdemir) [reply Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py]: that's pre-commit :)

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/blob.txt (hunk 1 above)
Comment (enes-oezdemir) [reply Packs/AzureStorageContainer/Integrations/AzureStorageContainer/blob.txt]: I have no idea, most likely pre-commit again removing it

File: Packs/AzureStorageContainer/ReleaseNotes/1_0_20.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+#### Integrations
+
+##### Azure Storage Container
+- Updated the Docker image to: *demisto/python3:3.10.13.86272*.
+
+- Fixed an issue where a **sas token** generated at container level didn't work properly.
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/ReleaseNotes/1_0_20.md:7]:
```suggestion
##### Azure Storage Container

- Updated the Docker image to: *demisto/python3:3.10.13.86272*.
- Fixed an issue where a **sas token** generated at container level didn't work properly.
```

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py [hunk 8]
```diff
@@ -1012,6 +1012,8 @@ def main() -> None:  # pragma: no cover
     global account_sas_token
     global storage_account_name
     account_sas_token = params.get('credentials', {}).get('password')
+    if account_sas_token and not account_sas_token.startswith("?"):
+        account_sas_token = "?" + account_sas_token
```
Comment (YuvHayun) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py:1016]:
```suggestion
        account_sas_token = f"?{account_sas_token}"
```

**Review by YuvHayun — APPROVED** [review]
Small comments.

**Comment by YuvHayun** [comment]
@enes-oezdemir  All looks good and ready to merge.
Please fill out the CLA so I can move on with the PR, you can find the link to it above.

**Pull request merged**